cryptoid/src/adapter.ts

```diff
@@ -0,0 +1,47 @@
+import { Requester, Validator } from '@chainlink/external-adapter'
+import { Config, ExecuteWithConfig, ExecuteFactory } from '@chainlink/types'
+import { makeConfig, DEFAULT_ENDPOINT } from './config'
+import { util } from '@chainlink/ea-bootstrap'
+
+const customParams = {
+  blockchain: ['blockchain', 'coin'],
+  endpoint: false,
+}
+
+const endpointToQ: { [key: string]: string } = {
+  difficulty: 'getdifficulty',
+  height: 'getblockcount',
+}
+
+export const execute: ExecuteWithConfig<Config> = async (request, config) => {
+  const validator = new Validator(request, customParams)
+  if (validator.error) throw validator.error
+
+  Requester.logConfig(config)
+
+  const jobRunID = validator.validated.id
+  const endpoint = validator.validated.data.endpoint || DEFAULT_ENDPOINT
+  const blockchain = validator.validated.data.blockchain.toLowerCase()
+
+  const key = util.getRandomRequiredEnv('API_KEY')
+  const q = endpointToQ[endpoint]
```

Comment:
What is `q`? Could it have a more explanatory name?

Reply:
q is a param of the cryptoid endpoint.

Reply:
I would give the var an explicit name, even if it will be a param of the request, but we'll know what is it.
```js
const someExplicitName = endpointToXXX[endpoint]
const params = { key, q: someExplicitName }
```